src/Schema/Comparator.php

```diff
@@ -506,6 +506,18 @@ public function diffColumn(Column $column1, Column $column2)
 
         // Null values need to be checked additionally as they tell whether to create or drop a default value.
         // null != 0, null != false, null != '' etc. This affects platform's table alteration SQL generation.
+        if ($properties1['default'] === 'NULL') {
+            $properties1['default'] = null;
```

Comment:
Most likely, this logic should be implemented as part of the schema introspection (Schema Manager, Platform), not as part of schema comparison. Otherwise, it will not be able to tell the difference between the string `"NULL"` and the `NULL` value.